include/rewd_controllers/MoveUntilTouchTopicController.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef REWD_CONTROLLERS_MOVEUNTILTOUCHTOPICCONTROLLER_HPP_
+#define REWD_CONTROLLERS_MOVEUNTILTOUCHTOPICCONTROLLER_HPP_
+
+#include <atomic>
+#include <mutex>
+#include <actionlib/server/action_server.h>
+#include <actionlib/client/simple_action_client.h>
+#include <hardware_interface/force_torque_sensor_interface.h>
+#include <hardware_interface/joint_command_interface.h>
+#include <pr_control_msgs/SetForceTorqueThresholdAction.h>
+#include <pr_control_msgs/TriggerAction.h>
+#include <pr_hardware_interfaces/TriggerableInterface.h>
+#include <geometry_msgs/WrenchStamped.h>
+#include <rewd_controllers/MultiInterfaceController.hpp>
+#include <rewd_controllers/JointTrajectoryControllerBase.hpp>
+
+namespace rewd_controllers
+{
+
+/// Uses a standard JointTrajectoryControllerBase and aborts the trajectory if
+/// the forces or torques are too big.
+/// Unlike MoveUntilTouchController, this controller gets its F/T data not
+/// via a hardware interface but rather via a ros topic.
+/// The other difference is, that taring is instigated through an action client
+/// instead of a hardware interface.
+class MoveUntilTouchTopicController final
+    : public MultiInterfaceController<hardware_interface::
+                                          PositionJointInterface,
+                                      hardware_interface::
+                                          VelocityJointInterface,
+                                      hardware_interface::EffortJointInterface,
+                                      hardware_interface::JointStateInterface>,
+      public JointTrajectoryControllerBase
+{
+
+public:
+  MoveUntilTouchTopicController();
+  ~MoveUntilTouchTopicController();
+
+  // Documentation inheried
+  bool init(hardware_interface::RobotHW* robot, ros::NodeHandle& n) override;
+
+  /** \brief This is called from within the realtime thread just before the
+   * first call to \ref update. It triggers a Tare on the controlled
+   * force/torque sensor hardware.
+   *
+   * \param time The current time
+   */
+  void starting(const ros::Time& time) override;
+
+  // Documentation inherited
+  void stopping(const ros::Time& time) override;
+
+  // Documentation Inherited.
+  void update(const ros::Time& time, const ros::Duration& period) override;
+
+protected:
+  /** \brief The JointTrajectoryControllerBase should accept new trajectories
```

Comment:
Why are we overriding this?

Reply:
Good point, no reason, just copied from `MoveUntilTouchController`.

Reply:
Let's remove it in that case. :)

Reply:
Oh, no, what I said was wrong. There actually is a reason and that is that the function is pure virtual in `JointTrajectoryControllerBase` and therefore needs to be overridden.